src/hooks/useWithdrawals.ts

```diff
@@ -1,27 +1,119 @@
+import { useCelo } from '@celo/react-celo';
 import { useCallback, useEffect, useState } from 'react';
 import { useContracts } from 'src/hooks/useContracts';
+import api from 'src/services/api';
 import { PendingWithdrawal } from 'src/types/account';
-import { Celo } from 'src/types/units';
+import { CeloWei } from 'src/types/units';
+
+export const useWithdrawalBot = (address: string | null) => {
+  const { managerContract, accountContract } = useContracts();
+
+  const finalizeWithdrawal = useCallback(async () => {
+    if (!address) return;
+    const [activeGroups, deprecatedGroups] = await Promise.all([
+      managerContract.methods.getGroups().call(),
+      managerContract.methods.getDeprecatedGroups().call(),
+    ]);
+    const groups = [...activeGroups, ...deprecatedGroups];
+    for (const group of groups) {
+      const scheduledWithdrawals = await accountContract.methods
+        .scheduledWithdrawalsForGroupAndBeneficiary(group, address)
+        .call();
+      if (scheduledWithdrawals === '0') continue;
+      await api.withdraw(address);
+      return;
+    }
+  }, [address, managerContract, accountContract]);
+
+  useEffect(() => {
+    void finalizeWithdrawal();
+    const intervalId = setInterval(finalizeWithdrawal, 120 * 1000);
+    return () => {
+      clearInterval(intervalId);
+    };
+  }, [finalizeWithdrawal]);
+};
+
+export const useClaimingBot = (address: string | null) => {
+  const { kit } = useCelo();
+  const { accountContract } = useContracts();
+
+  const claim = useCallback(async () => {
+    if (!address) return;
+    const { eth } = kit.connection.web3;
+
+    const [{ timestamp: currentBlockTimestamp }, { timestamps: withdrawalTimestamps }] =
+      await Promise.all([
+        eth.getBlock('latest'),
+        accountContract.methods.getPendingWithdrawals(address).call(),
+      ]);
+
+    const availableToClaim = !!(withdrawalTimestamps as string[]).find(
+      (withdrawalTimestamp) => withdrawalTimestamp < currentBlockTimestamp
+    );
+
+    if (availableToClaim) await api.claim(address);
+  }, [address, accountContract, kit.connection]);
+
+  useEffect(() => {
+    void claim();
+    const intervalId = setInterval(claim, 120 * 1000);
+    return () => {
+      clearInterval(intervalId);
+    };
+  }, [claim]);
+};
+
+/**
+ * Groups pending withdrawals that are within 5 minutes time span
+ */
+const fiveMinInSec: number = 5 * 60;
```

Comment:
Maybe better naming is what it does instead what is value of? Maybe sth like timestampThreshold?

Reply:
Done. Let me know what you think about the new name